lib/api_callable.js

```diff
@@ -168,6 +168,12 @@ function addTimeoutArg(aFunc, timeout, otherArgs, abTests) {
     options.deadline = new Date(now.getTime() + timeout);
     var metadata =
       otherArgs.metadataBuilder ? otherArgs.metadataBuilder(abTests) : null;
+    var headers = otherArgs.headers || {};
+    for (var key in headers) {
+      if (headers.hasOwnProperty(key)) {
+        metadata.set(key, headers[key]);
```

Comment:
I went  with simply excluding well-known headers since it seems to be the most simple solution.